src/QuantumNLDiffEq.jl

```diff
@@ -1,64 +1,137 @@
 module QuantumNLDiffEq
 
-import Yao: AbstractBlock, zero_state, expect, dispatch!, dispatch, chain
-import Flux: update!, ADAM, mse
-import Zygote: Buffer, gradient, ignore
-import SciMLBase: AbstractODEProblem, AbstractSciMLOperator
-	
+import Yao: AbstractBlock, zero_state, expect, dispatch!, dispatch, chain, Add, Scale, TimeEvolution, IdentityGate, igate
+import Flux: update!, Adam
+import Zygote: gradient
+import SciMLBase: AbstractODEProblem, AbstractSciMLOperator, ODEFunction
+import ForwardDiff: jacobian 
+import ChainRulesCore: rrule, NoTangent
+
+function rrule(func::ODEFunction, u, p, t)
+    y = func(u,p,t)
+    function func_pullback(ȳ)
+        Ju = jacobian((u)->func(u,p,t),u)
+        Jp = jacobian((p)->func(u,p,t),p)
+        NoTangent(), Ju'*ȳ, Jp'*ȳ, NoTangent()
+    end
+    return y, func_pullback
+end
```

Comment:
This kind of piracy is dangerous. Instead use a wrapper function and then define this on the wrapper.

Reply:
I guess `func::ODEFunction` is dangerous?
Thanks, will fix that!

Reply:
@ChrisRackauckas Does the latest commit fix that?